applications/rai-protocol.md

```diff
@@ -0,0 +1,168 @@
+We use bulletproofs to check the transaction value is valid, i.e. it is a
+non-negative number within the range [0, 2^n], and after the transaction the sender must still have a non-negative balance.
+
+The vanilla bulletproofs do not apply to the scenario of zether as Elgamal commitments
+are not fully homomorphic. We have tweaked a Bulletproofs library to support
+Sigma-protocols and also built a libproof library for substrate to use based on
+that tweaked Bulletproofs.
+Please see our [example code](https://github.com/RaiFinance/libproof).
+
+### RAI Runtime Modules
+
+RAI runtime modules will use LibProof to do confidentail transaction. For example,
+to create a verifiable confidential transaction, we implement method like `confidential::ConfidentialTransaction::create_transaction`, the resulting transaction can be verified with `confidential::ConfidentialTransaction::verify_transaction` method.
+
+### Substrate Client library
+
+polkadot.js based to interactive with ZKP based confidentail transaction and other RAI runtimes.
+
+## Ecosystem Fit
+
+RAI Finance improves the scalability of DeFi projects through secure off-chain transactions. By leveraging Zero-knowledge Proofs for trustless computation and cryptographic accumulators for immutable data storage, it is possible to provide a layer 2 solution that supports scalability, transparency, and privacy in transactions.
+
+The zkps also can be used on social trading feature on Rai Finance to protect strategies of traders(detailed history of transactions) while revealing the overall performance (e.g. profit rate, volatility, etc) of them.
+
+Though RAI will focus on Bulletproofs based proving system, other ZKP based projects can also reuse our LibProof and substrate client library to provide consistant user experience.
+
+## Team :busts_in_silhouette:
+
+### Team members
+* Name of team leader: 
+  * Keonho Lee
+* Names of team members: 
+  * Keonho Lee
+  * Taegyu Lim
+  * Myunghoon Kang
+  * Deokmun Kim
+  * Sandy Liang
+  * Minkyu Cho
+  * Hans Shi
+
+### Contact
+* **Contact Name:** Keonho Lee
+* **Contact Email:** keonho@rai.finance
+* Website: https://www.rai.finance
+
+### Legal Structure
+* **Registered Address:** 1 Irving Place #08-11 The Commerze@Irving Singapore 369546
+* **Registered Legal Entity:** RAI TECHNOLOGIES PTE. LTD.
+
+### Team's experience
+RAI Finance is led by a group of experts in cryptocurrency, trading and de-centralized finance. We have a technological and strategic partnership with the team who built innovative cryptocurrency trading platforms such as League of Traders2, a social trading platform that allows users to emulate the trading strategies of experienced and high profile traders and DEXEOS3, the first ever DEX (Decentralized Exchange) on EOS. Our goal is to simplify the trading process for retail users and bring new assets and liquidity into the DeFi Ecosystem.
+
+### Team Code Repos
+* https://github.com/RaiFinance
```

Comment:
why is the bulletproofs repo not marked as a fork of dalek's bulletproofs repo?

Reply:
I suppose libproof is not marked as a fork of https://github.com/suterusu-team/suter-proofs because it plans to differ radically.  Appears suter-proofs never understood what a SNARK is.  lol

Reply:
> linkedin profiles are inaccessible and largely irrelevant to assessing a team, so please include github profiles.
Will update.
> why is the bulletproofs repo not marked as a fork of dalek's bulletproofs repo?
Will do that.
> I suppose libproof is not marked as a fork of https://github.com/suterusu-team/suter-proofs because it plans to differ radically. Appears suter-proofs never understood what a SNARK is. lol
Absolutely not.

In short, this asking for grant proposal will focus on taking/integrate Bulletproofs into substrate/polkadot world as a different ZKP solution aim to extend the transaction scalability and privacy further more.